### PR TITLE
add interact river registry script

### PR DESCRIPTION
**Pull request by giuseppecrj**

### CODE REVIEW

File: contracts/scripts/interactions/InteractRiverRegistry.s.sol [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    chains[0] = 85432;
+    chains[1] = 11155111;
+    chains[2] = 550;
+    chains[3] = 6524490;
+    chains[4] = 8453;
+    chains[5] = 84532;
+    chains[6] = 137;
+    chains[7] = 42161;
+    chains[8] = 10;
+    chains[9] = 31337;
```
Comment (clemire) [inline contracts/scripts/interactions/InteractRiverRegistry.s.sol:26]: these two are anvil chains and may not be needed on alpha / gamma, etc

File: contracts/scripts/interactions/InteractRiverRegistry.s.sol [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+// contracts
+import {Interaction} from "contracts/scripts/common/Interaction.s.sol";
+
+contract InteractRiverRegistry is Interaction {
+  function __interact(address deployer) internal override {
+    address riverRegistry = getDeployment("riverRegistry");
+
+    uint64[] memory chains = new uint64[](13);
+    chains[0] = 85432;
```
Comment (clemire) [inline contracts/scripts/interactions/InteractRiverRegistry.s.sol:17]: Can you put a comment after the chain ids with the network they reference?

e.g.

```
chains[0] = 85432/ // Base Sepolia
...
chains[8] = 10; // Gnosis Mainnet
```

File: contracts/scripts/interactions/InteractRiverRegistry.s.sol [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    chains[7] = 42161;
+    chains[8] = 10;
+    chains[9] = 31337;
+    chains[10] = 31338;
+    chains[11] = 100;
+    chains[12] = 10200;
+
+    vm.startBroadcast(deployer);
+    IRiverConfig(riverRegistry).setConfiguration(
+      "xchain.blockchains",
```
Comment (clemire) [inline contracts/scripts/interactions/InteractRiverRegistry.s.sol:33]: Doesn't this parameter need to be the keccak hash of "xchain.blockchains"?

Comment (giuseppecrj) [reply contracts/scripts/interactions/InteractRiverRegistry.s.sol]: just need to typecase

File: contracts/scripts/interactions/InteractRiverRegistry.s.sol (hunk 2 above)
Comment (giuseppecrj) [reply contracts/scripts/interactions/InteractRiverRegistry.s.sol]: fixed in latest push

File: contracts/scripts/interactions/InteractRiverRegistry.s.sol (hunk 1 above)
Comment (giuseppecrj) [reply contracts/scripts/interactions/InteractRiverRegistry.s.sol]: fixed

File: contracts/scripts/interactions/InteractRiverRegistry.s.sol [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    chains[5] = 84532; // base sepolia
+    chains[6] = 137; // polygon
+    chains[7] = 42161; // arbitrum
+    chains[8] = 10; // optimism
+    chains[9] = 100; // gnosis
+    chains[10] = 10200; // gnosis chiado
+
+    vm.startBroadcast(deployer);
+    IRiverConfig(riverRegistry).setConfiguration(
+      bytes32("xchain.blockchains"),
```
Comment (clemire) [inline contracts/scripts/interactions/InteractRiverRegistry.s.sol:31]: So in the script I wrote, I followed the existing pattern for other config keys and i keccak-encoded `xchain.blockchains`, see https://github.com/river-build/river/blob/90fc274e6d832e2cc71eb0ed84e08f9d3de7a85e/scripts/set-riverchain-config.sh#L32

Can you keccak encode this string and use the resulting bytes32 as the hashed key?

Comment (giuseppecrj) [reply contracts/scripts/interactions/InteractRiverRegistry.s.sol:31]: why are you encoding it?

Comment (giuseppecrj) [reply contracts/scripts/interactions/InteractRiverRegistry.s.sol:31]: it takes less bytes to just typecase than to encode it

Screenshot 2024-09-19 at 12 11 57 PM

Comment (giuseppecrj) [reply contracts/scripts/interactions/InteractRiverRegistry.s.sol:31]: nvm, realized `xchain.blockchains` is the actual key name and not a namespace path

Comment (giuseppecrj) [reply contracts/scripts/interactions/InteractRiverRegistry.s.sol]: updated in latest push

**Review by clemire — APPROVED** [review]

**Pull request merged**